numpy/doc/subclassing.py

```diff
@@ -555,6 +555,46 @@ def __array_wrap__(self, arr, context=None):
 how this can work, have a look at the ``memmap`` class in
 ``numpy.core``.
 
+Subclassing and Downstream Compatibility
+----------------------------------------
+
+When sub-classing ``ndarray`` or creating duck-types that mimic the ``ndarray``
+interface, it is your responsibility to decide how aligned your APIs will be
+with those of numpy. There is sometimes great temptation and confusion amongst
```

Comment:
The second sentence in this paragraph reads as too judgmental to me. I would drop it.

It would be good to summarize NumPy's approach here in addition to giving the sum example, e.g.,

     For convenience, many numpy functions that have a corresponding ndarray
     method (e.g., sum, mean, take, reshape) work by checking if the first argument
     to a function has a method of the same name. If it exists, the method is called
     instead of coercing the arguments to a numpy array.